hail_search/hail_search_query.py

```diff
@@ -142,42 +143,46 @@ def __init__(self, data_type, sample_data, genome_version, sort=XPOS, num_result
         self._genome_version = genome_version
         self._sort = sort
         self._num_results = num_results
+        self._data_type = data_type
         self._ht = None
         self._enums = None
         self._globals = None
 
-        self._load_filtered_table(data_type, sample_data, **kwargs)
+        self._load_filtered_table(sample_data, **kwargs)
 
-    def _load_filtered_table(self, data_type, sample_data, intervals=None, exclude_intervals=False, variant_ids=None, **kwargs):
+    def _load_filtered_table(self, sample_data, intervals=None, exclude_intervals=False, variant_ids=None, **kwargs):
         parsed_intervals, variant_ids = self._parse_intervals(intervals, variant_ids)
         excluded_intervals = None
         if exclude_intervals:
             excluded_intervals = parsed_intervals
             parsed_intervals = None
+        self._load_table_kwargs = {'_intervals': parsed_intervals, '_filter_intervals': bool(parsed_intervals)}
         self.import_filtered_table(
-            data_type, sample_data, intervals=parsed_intervals, excluded_intervals=excluded_intervals,
-            variant_ids=variant_ids, **kwargs)
+            sample_data, excluded_intervals=excluded_intervals, variant_ids=variant_ids, **kwargs)
 
-    def import_filtered_table(self, data_type, sample_data, intervals=None, **kwargs):
-        tables_path = f'{DATASETS_DIR}/{self._genome_version}/{data_type}'
-        load_table_kwargs = {'_intervals': intervals, '_filter_intervals': bool(intervals)}
+    def _get_table_path(self, path):
+        return f'{DATASETS_DIR}/{self._genome_version}/{self._data_type}/{path}'
 
+    def _read_table(self, path):
```

Comment:
If we want a cache over our table reads, maybe https://cachetools.readthedocs.io/en/latest/#cachetools.cachedmethod?

Reply:
its not really caching, its assigning it to an instance variable. This is essentially the same thing we are doing when we say `self._ht = ...` but I tried to make it a little more generic to make it extensible for some other known cases. I'll refactor this to make it a clearer case of reuse though, since this seems to be really confusing

Reply:
also these tables live in memory, so from a performance standpoint I'm not sure caching helps. This is juts a helper function to make reading multiple tables with the same arguments easier, but its a different table for each call

Reply:
the first above comment was because I thought you were commenting on a different part of the code. The second comment I think actually responds to your comment

Reply:
oh so then my comment above does basically work - its not a cache its an instance variable, I just know theres gonna be at least 1 more so I assigned it to a dictionary instead of its own variable